### ISSUE TITLE
Enable Path to `ranlib` to be Provided by Environment Variable (Similar to `LD` and `OBJCOPY` Env Vars)

### ISSUE THREAD

**Issue by GuyEP-Modulate** [tux3/armerge#7]
Could we get support for the path to `ranlib` being supplied via an environment variable (perhaps, `RANLIB`), like is possible with `LD` and `OBJCOPY`?

### Background Information
In attempting to create a static library for a project that targets Android NDK 25.2.9519653 and higher, I've found that the symbols in a static library that was output by `armerge` are not recognized; all the symbols show up as undefined during linking of the project that uses the static library. For example, pretending my library is called `myproject` and that we have a `get_version` and `global_init` functions inside, we get this when trying to link `myproject_freestanding.a` (the output from `armerge`):
```
Error: undefined symbol: myproject_get_version
Error: undefined symbol: myproject_global_init
```

After a lot of trial and error, I discovered that the errors seem to be because `armerge` is using the system-supplied version of `ranlib` instead of the version that ships with the NDK. I am on macOS, and the issue is that the system-supplied version (in `/usr/bin/ranlib`) is either not making any changes to the static library table of contents or is making a change that makes the library incompatible with the NDK-supplied clang linker.

As a workaround, I have put a symlink to `/Users/$USER/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/bin/llvm-ranlib` at `~/bin/ranlib` and put `~/bin` first in my path so that `armerge` uses this version instead of the system-supplied `ranlib`.

As an aside: I also have to specify `ARMERGE_LDFLAGS` of `--target=aarch64-none-linux-android33` or I will get an error like this (our project uses both libOpus and libOpsEnv as dependencies):
```
stdout:
stderr: ld64.lld: warning: Option `-r' is not yet implemented. Stay tuned...
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@tables_pitch_lag.o.Qy7T5T1V.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@check_control_input.o.b1c9GNxp.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@NLSF_decode.o.FXkr22gt.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@apply_sine_window_FLP.o.bw8TPdZI.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopusenc.a@resample.o.Qf9Buucr.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@mathops.o.yV3vY3Za.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@init_encoder.o.dbbAIkgr.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@LPC_analysis_filter_FLP.o.MOfvM78I.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@celt_neon_intr.o.sF04cHqx.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@LPC_inv_pred_gain.o.kKcef5fc.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@lin2log.o.8vk6nSmv.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@decode_core.o.wjkWJTIc.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopusenc.a@ogg_packer.o.aLcamJ6j.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@encode_pulses.o.DDg5s7XQ.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@pitch.o.hp4GOCJC.o: unhandled file type
ld64.lld: error: /var/folders/9w/qjb_h_r54jsf039kwm_gnc3w0000gp/T/armerge.IsJ6pH/libopus.a@resampler_private_up2_HQ.o.BRyuANYX.o: unhandled file type
ld64.lld: error: too many errors emitted, stopping now (use --error-limit=0 to see all errors)
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

Here's an example of a command line that works for me (assuming a symlink in `~/bin/ranlib` to `/Users/$USER/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/bin/llvm-ranlib`):
```
PATH=/Users/$USER/bin:$PATH \
LD=/Users/$USER/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang++ \
ARMERGE_LDFLAGS="--target=aarch64-none-linux-android33" \
OBJCOPY=/Users/guy/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/bin/llvm-objcopy \
armerge --output libmyproject_freestanding.a \
  libopus.a \
  libopusenc.a \
  libmyproject.a \
  --keep-symbols 'myproject_*'
```

**Comment by tux3** [comment]
Hey, thanks for the suggestion! Seems good, and simple enough to add =)
I've pushed a branch with the feature, could you please try it out with `cargo install --git https://github.com/tux3/armerge --branch ranlib` and check that it works as expected for you?  
I don't expect any problems, but I've also bundled some maintenance dependency updates while I'm at it, and you can never be too careful.

Also about the flags you're using, have you tried setting `LD` to  `<....>/prebuilt/darwin-x86_64/bin/ld.lld` instead of setting it to `clang++`?  
We build all our Android stuff inside Dockers, but I think this should work even on Mac without needing to give the NDK an  extra target argument in LDFLAGS. We do `LD=${ANDROID_BIN_PATH}/ld.lld` and  `OBJCOPY=${ANDROID_BIN_PATH}/llvm-objcopy`, which works without having to pass the --target argument in our case (although we are building with ndk r26c)

**Comment by tux3** [comment]
I should also see if we can do a better job of detecting when the system `ranlib` is not doing its job to avoid silently producing broken files.  
Maybe we could just reject Apple ranlib outright. Even for native Mac builds we fallback to Apple's libtool instead.

**Comment by GuyEP-Modulate** [comment]
WOW! I will give this a spin! I really appreciate how quickly you turned this around; thank you.

**Comment by GuyEP-Modulate** [comment]
@tux3 This appears to work well! Thanks, again. And, thank you as well for the hint regarding the path to `LD`; I was actually able to use `/Users/$USER/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/bin/ld`, which appears to be a symlink to `/Users/$USER/Library/Android/sdk/ndk/25.2.9519653/toolchains/llvm/prebuilt/darwin-x86_64/bin/ld.lld`. This allowed me to drop ARMERGE_LDFLAGS completely.

**Comment by tux3** [comment]
Thanks for testing, I've released a v2.1.0 =)
